### PR TITLE
Add build recipe for Arrow HEAD commit, 0.1 pre-release

### PR DESCRIPTION
**Pull request by wesm**
We've inverted the dependency relationship between Arrow and Parquet, so I'll update the parquet-cpp recipe, and this will need to become feedstock before parquet-cpp can

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@jakirkham @ericdill Since this is part of 3 heavily interrelated libraries (#1732 and #992), do we want to just do all three in the same PR?  I remember that's what I did with cyavro.

**Comment by wesm** [comment]
Please let me know what's the most expedient route to merging these three recipes -- while I've tested locally there may be follow up work to get the chain working.

File: recipes/arrow-cpp/build.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+cmake \
+    -DCMAKE_BUILD_TYPE=release \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DARROW_BUILD_BENCHMARKS=off \
+    -DARROW_BUILD_TESTS=off \
+    -DARROW_HDFS=on \
+    ..
+
+make -j4
```
Comment (ocefpaf) [inline recipes/arrow-cpp/build.sh]: Do not hardcode the `4`. Use `${CPU_COUNT}` instead.

**Comment by ocefpaf** [comment]
Looks good. Just a few minor comments.

**Comment by chrisaycock** [comment]
LGTM

**Comment by wesm** [comment]
Looks like Travis hiccupped after the merge

**Comment by ocefpaf** [comment]
> Looks like Travis hiccupped after the merge

The feedstock creation should happen. I will look into the Travis-CI errors later today. Let me know if you do not see the feedstock though.

**Pull request merged**